include/swift/Driver/ToolChain.h

```diff
@@ -87,12 +87,12 @@ class ToolChain {
 
   /// Packs together information chosen by toolchains to create jobs.
   struct InvocationInfo {
-    const char *ExecutableName;
+    StringRef ExecutableName;
```

Comment:
No, we've gotten into trouble here before. No StringRef should ever be assumed null-terminated. What's wrong with `char *`?